### DISCUSSION TITLE
cognition.run website got stuck

### DISCUSSION THREAD

**Discussion by mingjia1025** [jspsych/jsPsych#1365] [General]
@javidalpe @jodeleeuw 

When I'm editing my jsPsych code on cognition.run (on the source code page), the page suddenly got frozen. I tried to reload the page but it just failed to be loaded with the error code "out of memory". I tried different web browsers but still to no avail.... 

By the way, I tried to duplicate some of my previous experiment codes and they worked fine at first. But when I tried to make certain changes to the code, these new tasks also became dead. 

Are the cognition.run website running into any technical issues?

Thanks

**Comment by jodeleeuw** [comment]
Hi @mingjia1025 

What changes did you make to the code before the tasks died?

**Comment by javidalpe** [comment]
Hello @mingjia1025. We are not aware of an issue regarding code editor.

In order to help you, could you please:
- Send us the code of the experiment or the link to the experiment.
- Send us the "certain changes" that you introduce to the working tasks?
- Try the experiment in your local machine. Does it become dead?

**Reply by mingjia1025** [reply]
The link to my experiment is: https://hscusrrats.cognition.run

In the attached code, I tried to made a copy of the code block names "construct training stimuli" (lines 75-170) right after it, then change all the variables "i_train" and "n_train" to "i_test" and "n_test" (defined at this point of the experiment). The task died when I changed "i_train++" to "i_test++" in the for-loop control statement without changing anything inside the loop.

I'm not sure how to try the experiment on my local machine. I've always been hosting the experiment using cognition.run and I don't know anyway to run it offline. 
[test code.txt](https://github.com/jspsych/jsPsych/files/5793312/test.code.txt)

**Reply by javidalpe** [reply]
The CPU of my machine goes to 100% after loading your experiment, which means there is an infinite loop in the code. This also explain why the editor get frozen: the browser is executing the infinite loop and the UI is blocked.

I have found at least one infinite for loop at line 187:
```
for (i_test = 0; i_test < n_train; i_train++){
```

If you don't update `i_test` variable, `i_test` will be always lower than `n_train` and the condition will be always true. The body of the loop will run again and again and again ;)

**Reply by mingjia1025** [reply]
Thank you for your answer. I see what you mean now. Is there a way to stop the editor from evaluating the code on the fly? You know it's a bit inconvenient for a non-professional coder to fiddle with code while keeping in mind certain technical things that would make the code frozen.

**Reply by javidalpe** [reply]
Unfortunately no. I'll investigate how to capture this type of issues that makes the task unavailable to edit again, or create a new feature to edit the source code without a live preview.
